widgets/nuxeo-directory-widget-behavior.html

```diff
@@ -0,0 +1,147 @@
+<link rel="import" href="../nuxeo-i18n-behavior.html">
+<link rel="import" href="../../iron-form-element-behavior/iron-form-element-behavior.html">
+<link rel="import" href="../../iron-validatable-behavior/iron-validatable-behavior.html">
+
+<script>
+  var Nuxeo = Nuxeo || {};
+  (function () {
+
+    /**
+     * `Nuxeo.DirectoryWidgetBehavior` provides helpers to build a widget to select Nuxeo vocabulary entrie(s).
+     *
+     * @polymerBehavior Nuxeo.DirectoryWidgetBehavior
+     */
+    Nuxeo.DirectoryWidgetBehavior = [Nuxeo.I18nBehavior, Polymer.IronFormElementBehavior, Polymer.IronValidatableBehavior, {
+      properties: {
+        /**
+         * Name of the directory.
+         */
+        directoryName: {
+          type: String
+        },
+
+        /**
+         * Checking this option means that the labels are localized with translations provided
+         * in the directory itself (i.e. in fields). Otherwise labels are translated as usual
+         * picking values in messages*.properties files.
+         */
+        dbl10n: {type: Boolean, value: false},
+
+        /**
+         * Label.
+         */
+        label: String,
+
+        /**
+         * In case of hierarchical vocabulary, if true, parent item can be selected.
+         */
+        canSelectParent: Boolean,
+
+        /**
+         * Set to `true` for read only mode.
+         */
+        readonly: {
+          type: Boolean,
+          value: false
+        },
+
+        /**
+         * Function used to get the id from the choice object.
+         */
+        idFunction: {
+          type: Function,
+          value: function() {
+            return this._idFunction.bind(this);
+          }
+        },
+
+        /**
+         * Error message to show when `invalid` is true.
+         */
+        errorMessage: String,
+
+        /**
+         * Formatter for the entries.
+         */
+        format: {
+          type: Function,
+          value: function() {
+            return this._formatter.bind(this);
+          }
+        },
+
+        /**
+         * Fired when the directory entries are loaded.
+         *
+         * @event loaded
+         */
+
+        _entries: Array
+      },
+
+      observers: [
+        '_fetchEntries(directoryName)'
+      ],
+
+      _fetchEntries: function() {
+        if (this.directoryName) {
+          this.value = [];
+          this.async(function() {
+            this.$.op.params = {
+              'directoryName': this.directoryName,
+              'dbl10n': this.dbl10n,
+              'canSelectParent': this.canSelectParent,
+              'localize' : true,
+              'lang': (window.nuxeo.I18n.language) ? window.nuxeo.I18n.language.split('-')[0] : 'en'
+             };
+            this.$.op.execute().then(function(resp) {
+              this._entries = [];
+              resp.forEach(function(entry) {
+                this._populate(entry);
+              }.bind(this));
+              this.fire('loaded');
```

Comment:
should we call it something more specific? `directory-entries-loaded` or something?